bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractMockupCacheWorker.php

```diff
@@ -70,6 +70,10 @@ public function saveToMockupCache($objectId, $data = null)
         if (empty($data)) {
             $data = $this->db->fetchOne('SELECT data FROM ' . $this->getStoreTableName() . ' WHERE o_id = ? AND tenant = ?', [$objectId, $this->name]);
             $data = json_decode($data, true);
+            $worker = $this->getTenantConfig()->getTenantWorker();
+            if(method_exists($worker, "doPreIndexDataModification")) {
```

Comment:
Should we introduce an interface for that?

Reply:
I though about something like that as well, another option would be to extract line 71 and 72 in a new method `getDataForMockupCache` or somthing like that and override it in the AbstractElasticSearch Worker? 
In that way we could (now or later) tackle the other idea as well, and fetch the data directly from ES.